loaders/embededKeyLoader.go

```diff
@@ -0,0 +1,133 @@
+package loaders
+
+import (
+	"embed"
+	"fmt"
+	"sync"
+
+	"github.com/iden3/go-circuits/v2"
+	"github.com/pkg/errors"
+)
+
+//go:embed verification_keys/*.json
+var defaultKeys embed.FS
+
+// ErrKeyNotFound is returned when key is not found
+var ErrKeyNotFound = errors.New("key not found")
+
+// EmbeddedKeyLoader load keys from embedded FS or filesystem.
+// Filesystem has priority if keyLoader specified.
+type EmbeddedKeyLoader struct {
+	keyLoader VerificationKeyLoader
+	cache     map[circuits.CircuitID][]byte
+	cacheMu   sync.RWMutex
```

Comment:
https://pkg.go.dev/sync#RWMutex
```go
A RWMutex must not be copied after first use.
```
Use pointer to sync.RWMutex to prevent copy in methods.